docs/guides/install-c-api.mdx

```diff
@@ -0,0 +1,158 @@
+
+You can check everything is installed via
+```bash
+rustc --version
+gcc --version
+cbindgen --version
+make --version  # optional, but recommended
+```
+
+### Building
+
+To build the C header and library, you can run the following Make command[^1] in Qiskit root,
+[^1]: If you did not install Make, check the `Makefile` in Qiskit root for the required commands. (Or just install Make, it's not too late.)
+```bash
+make c
+```
+which will provide the compiled shared library in `dist/c/lib` and the `qiskit.h` header with
+all function declarations in `dist/c/include`. Note that the precise library name depends
+on the platform, e.g `libqiskit.so` on Unix and `libqiskit.dylib` on MacOS.
+(Note that this step currently emits a lot of warnings. This is fine, we're working on it.)
+
+We can then compile a C program using the Qiskit C header and library, by
+```bash
+gcc example.c -o example.o -I /path/to/dist/c/include -L /path/to/dist/c/lib -lqiskit
+```
+To ensure the library is found during execution, set the runtime library path to
+include `/path/to/dist/c/lib`. This depends on the platform, e.g. on Linux
+```bash
+export LD_LIBRARY_PATH=/path/to/dist/c/lib:$LD_LIBRARY_PATH
+```
+or on MacOS
+```bash
+export DYLD_LIBRARY_PATH=/path/to/dist/c/lib:$DYLD_LIBRARY_PATH
+```
+Alternatively, we can set the runtime library path during compilation by adding
+```
+-Wl,-rpath,/path/to/dist/c/lib
+```
+to the compiler flags.
+
+Now we can execute the binary,
+```bash
+./example.o
+```
+which, if using the example snippet shown above, should print
+```
+num_qubits: 100
+num_terms: 1
+observable: SparseObservable { num_qubits: 100, coeffs: [Complex { re: 2.0, im: 0.0 }], bit_terms: [X, Y, Z], indices: [0, 1, 2], boundaries: [0, 3] }
+```
+
+## Windows (experimental)
+
+<Admonition type="caution">
+  Windows support is currently experimental. Instructions are work in progress.
+</Admonition>
+
+### Requirements
+
+We require the following tools for the compilation:
+* A Rust compiler: see e.g. the [guide on installing Qiskit from source](https://github.com/Qiskit/qiskit/blob/main/CONTRIBUTING.md#installing-qiskit-from-source).
+* A C compiler: we'll use MSVC and the native command prompt offering the ``cl`` command
+* A Python installation, both with access to ``python3.lib`` and ``python3.dll``.
+* [`cbindgen`](https://github.com/mozilla/cbindgen): a tool to crate the C header, which you can e.g. install via `cargo install cbindgen`.
+    Note that running the tool from the command line should be enabled, which might require exporting your `PATH` variable
+    to include `/path/to/.cargo/bin`.
+
+### Building
+
+First, we compile the `qiskit_cext` dynamic library, by running the following in Qiskit root
+```bash
+set PATH="\path\to\pythonlib\";%PATH%
+cargo rustc --release --crate-type cdylib -p qiskit-cext
+```
+This will generate the ``.dll`` dynamic library, and associated ``.dll.lib`` file, in `target/release`.
+Next, we have to generate the header via
```

Comment:
```suggestion
This will generate the ``.dll`` dynamic library and associated ``.dll.lib`` file in `target/release`.
Next, generate the header with
```